Src/Newtonsoft.Json.UnityConverters/QuaternionConverter.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+using UnityEngine;
+
+namespace Newtonsoft.Json.UnityConverters
+{
+    public class QuaternionConverter : JsonConverter
+    {
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            var qt = (Quaternion)value;
+            writer.WriteStartObject();
+            writer.WritePropertyName("w");
+            writer.WriteValue(qt.w);
+            writer.WritePropertyName("x");
+            writer.WriteValue(qt.x);
+            writer.WritePropertyName("y");
+            writer.WriteValue(qt.y);
+            writer.WritePropertyName("z");
+            writer.WriteValue(qt.z);
+
+            writer.WritePropertyName("eulerAngles");
+            writer.WriteStartObject();
+            writer.WritePropertyName("x");
+            writer.WriteValue(qt.eulerAngles.x);
+            writer.WritePropertyName("y");
+            writer.WriteValue(qt.eulerAngles.y);
+            writer.WritePropertyName("z");
+            writer.WriteValue(qt.eulerAngles.z);
+            writer.WriteEndObject();
+
+            writer.WriteEndObject();
+        }
+
+        public override bool CanConvert(Type objectType)
+        {
+            return objectType == typeof(Quaternion);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            var obj = JObject.Load(reader);
+            var props = obj.Properties().ToList();
+
+            var result = new Quaternion();
+            if (props.Any(p => p.Name == "w"))
+                result.w = (float)obj["w"];
+
+            if (props.Any(p => p.Name == "x"))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity.Converters/pullRequest?prid=4846771)